src/Mod/PartDesign/App/FeatureChamfer.cpp

```diff
@@ -104,6 +104,12 @@ short Chamfer::mustExecute() const
 
 App::DocumentObjectExecReturn *Chamfer::execute()
 {
+    if (onlyHasToRefine()){
+        TopoShape result = refineShapeIfActive(rawShape);
+        Shape.setValue(result);
+        return App::DocumentObject::StdReturn;
+    }
+

```

Comment:
This code block is used ten times in various sub-classes. The code should go to a helper method `onlyHaveRefined` in the base class and then used as
```
if (onlyHaveRefined()) {
    return App::DocumentObject::StdReturn;
}
```